plugins/src/main/java/org/smartregister/fhir/gateway/plugins/SyncAccessDecision.java

```diff
@@ -101,8 +101,25 @@ public RequestMutation getRequestMutation(RequestDetailsReader requestDetailsRea
         if (isSyncUrl(requestDetailsReader)
                 && !shouldSkipDataFiltering(
                         requestDetailsReader)) { // Check if it is the Sync URL and Skip app-wide
+
             // accessible resource requests
-            if (syncStrategyIds.isEmpty()
+            if (Constants.RELATED_ENTITY_LOCATION.equalsIgnoreCase(syncStrategy)) {
+                Map<String, String[]> parameters =
+                        new HashMap<>(requestDetailsReader.getParameters());
+                String[] syncLocations = parameters.get(Constants.SYNC_LOCATIONS);
+                List<String> relatedEntityLocationIds;
+                if (roles.contains(Constants.ROLE_ALL_LOCATIONS) && syncLocations != null) {
+                    // selected locations
+
+                    List<String> locationUuids = Arrays.asList(syncLocations[0].split(","));
+                    relatedEntityLocationIds =
+                            PractitionerDetailsEndpointHelper.getAttributedLocations(
+                                    PractitionerDetailsEndpointHelper.getLocationsHierarchy(
+                                            locationUuids));
+                    this.syncStrategyIds = Map.of(syncStrategy, relatedEntityLocationIds);
+                }
+
+            } else if (syncStrategyIds.isEmpty()
```

Comment:
This should be a separate if statement _(rather than else if)_ so that incase the newly processed `this.syncStrategyIds` is empty the exception is still thrown and logged.